plugins/riscv/riscv_main.ml

```diff
@@ -0,0 +1,479 @@
+    | "LW", [|r0;r1;off|] -> mem_type_load load W Signed r0 r1 off
+    | "LH", [|r0;r1;off|] -> mem_type_load load H Signed r0 r1 off
+    | "LHU", [|r0;r1;off|] -> mem_type_load load H Unsigned r0 r1 off
+    | "LB", [|r0;r1;off|] -> mem_type_load load B Signed r0 r1 off
+    | "LBU", [|r0;r1;off|] -> mem_type_load load B Unsigned r0 r1 off
+    | "SW", [|r0;r1;off|] -> mem_type_store store W r0 r1 off
+    | "SH", [|r0;r1;off|] -> mem_type_store store H r0 r1 off
+    | "SB", [|r0;r1;off|] -> mem_type_store store B r0 r1 off
+    | _ -> Ok [Bil.special (Insn.asm insn)]
+
+  (** Branching instructions *)
+  let beq mem rs1 rs2 dst = Bil.[
+    Bil.if_ (!$rs1 = !$rs2) [
+      Bil.Jmp (Bil.int (Memory.min_addr mem) + !%dst);
+    ][];
+  ]
+
+  let bne mem rs1 rs2 dst = Bil.[
+    Bil.if_ (!$rs1 <> !$rs2) [
+      Bil.Jmp (Bil.int (Memory.min_addr mem) + !%dst);
+    ][];
+  ]
+
+  let bge mem rs1 rs2 dst = Bil.[
+    Bil.if_ (!$rs1 > !$rs2) [
+      Bil.Jmp (Bil.int (Memory.min_addr mem) + !%dst);
+    ][];
+  ]
+
+  let blt mem rs1 rs2 dst = Bil.[
+    (* Bil.if_ (signed rs1 < signed rs2) [ *)
+    Bil.if_ (!$rs1 < !$rs2 ) [
+      Bil.Jmp (Bil.int (Memory.min_addr mem) + !%dst);
+    ][];
+  ]
+
+  let jalr mem rs rd off = Bil.[
+    rd := Bil.int (Memory.min_addr mem) + Bil.int (Word.of_int 4 ~width:32);
+    Bil.Jmp (!$rs + !%off);
+  ]
+
+  let jal mem rd off = Bil.[
+    rd := Bil.int (Memory.min_addr mem) + Bil.int (Word.of_int 4 ~width:32);
+    Bil.Jmp (!%off);
+  ]
+
+  let lift_branch mem insn = match Insn.name insn, Insn.ops insn with
+    | "BEQ", [|r0;r1;addr|] -> bra_type beq mem r0 r1 addr
+    | "BNE", [|r0;r1;addr|] -> bra_type bne mem r0 r1 addr
+    | "BGE", [|r0;r1;addr|] -> bra_type bge mem r0 r1 addr
+    | "BLT", [|r0;r1;addr|] -> bra_type blt mem r0 r1 addr
+    | "BLTU", [|r0;r1;addr|] -> bra_type blt mem r0 r1 addr
+    | "BGEU", [|r0;r1;addr|] -> bra_type bge mem r0 r1 addr
+    | "JAL", [|r0;addr|] -> bra_type2 jal mem r0 addr
+    | "JALR", [|r0;r1;off|] -> bra_type jalr mem r0 r1 off
+    (* | "RET", [||] -> rrr_type ret *)
+    | _ -> Ok [Bil.special (Insn.asm insn)]
+
+(*
+  let lift_csr mem insn = match Insn.name insn, Insn.ops insn with
+    | "CSRRW", [|r0;r1|] -> rr_type csrrw r0 r1 r1
+    | "CSRRS", [|r0;r1|] -> rr_type csrrs r0 r1 r1
+    | "CSRRC", [|r0;r1|] -> rr_type csrrc r0 r1 r1
+    | "CSRRWI", [|r0;imm|] -> ri_type csrrw r0 imm
+    | "CSRRSI", [|r0;imm|] -> ri_type csrrs r0 imm
+    | "CSRRCI", [|r0;imm|] -> ri_type csrrc r0 imm
+    | _ -> Ok [Bil.special (Insn.asm insn)]
+*)
+
+  let riscv_ops_exn ops () =
+    Array.map (ops) ~f:(fun op ->
+      Option.value_exn
+        ~here:[%here]
+        ~error:(Error.create "unsupported operand" op Op.sexp_of_t )
+        (Riscv_op.create op))
+
+  let riscv_ops ops = try_with (riscv_ops_exn ops)
+
+
+  (** Substitute PC with its value  *)
+  let resolve_pc mem = Stmt.map (object(self)
```

Comment:
apparently you're not using the `pc` register anywhere, thus you don't need this pass. If you will remove the `pc` register from the scope it all, it will on strengthen this position.